### PR TITLE
java PG: fix detection of arm

### PR DESCRIPTION
**Pull request by catap**
#### Description

`${configure.build_arch}` and `[option configure.build_arch]` both return an empty string here.

This lead to using default fallback on clean system which is `openjdkXY` which probably won't be build on arm64.

To fix it I've migrated to `${os.arch}` which is available and which has value `arm` instead of `arm64`.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

I've tested it on clean

macOS 12.4 21F79 arm64
Xcode 13.1 13A1030d

by simple installing `scala3.2` port.

###### Verification 
Have you

- [ ] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [ ] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [ ] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by catap** [comment]
@breun, @usersxx as far as I recall you're the one who is working on java and related things this days.

So, I'd love to ask your review.

File: _resources/port1.0/group/java-1.0.tcl [hunk 1]
```diff
@@ -59,7 +59,9 @@ namespace eval java {
             # as required, as currently these are the only ones supporting arm.
             # To be reviewed as support for arm comes for the other versions.
             # Following regex matches openjdk<version> only.
-            if { [option configure.build_arch] eq "arm64" &&
+            # Keep in mind that configure.build_arch doesn't available here
```
Comment (breun) [inline _resources/port1.0/group/java-1.0.tcl:62]:
```suggestion
            # Keep in mind that configure.build_arch isn't available here
```

**Comment by catap** [comment]
How it works right now:
```
catap@Kirills-mini-m1 ~ % port -d deps scala3.2
DEBUG: Changing to port directory: /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/lang/scala
DEBUG: OS darwin/21.5.0 (macOS 12.4) arch arm
DEBUG: Sourcing PortGroup github 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/github-1.0.tcl
DEBUG: Sourcing PortGroup java 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/java-1.0.tcl
DEBUG: Sourcing PortGroup deprecated 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/deprecated-1.0.tcl
DEBUG: Re-registering default for livecheck.url
DEBUG: Re-registering default for livecheck.regex
DEBUG: Sourcing PortGroup select 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/select-1.0.tcl
DEBUG: universal_variant is false, so not adding the default universal variant
DEBUG: Running callback portconfigure::add_automatic_compiler_dependencies
DEBUG: Finished running callback portconfigure::add_automatic_compiler_dependencies
DEBUG: Running callback portbuild::add_automatic_buildsystem_dependencies
DEBUG: Finished running callback portbuild::add_automatic_buildsystem_dependencies
DEBUG: Running callback portstartupitem::add_notes
DEBUG: Finished running callback portstartupitem::add_notes
DEBUG: Running callback java::java_set_env
DEBUG: java-portgroup: Trying to find JVM version: 1.8+
DEBUG: No value for java JAVA_HOME was automatically discovered
DEBUG: Adding dependency on JDK fallback openjdk8
DEBUG: Finished running callback java::java_set_env
DEBUG: Running callback deprecated.deprecate_port
DEBUG: Finished running callback deprecated.deprecate_port
Full Name: scala3.2 @3.2.0_1
Library Dependencies: port:scala_select, port:openjdk8
DEBUG: Checking time since last reclaim run
catap@Kirills-mini-m1 ~ % 
```
after the patch is applied
```
catap@Kirills-mini-m1 ~ % port -d deps scala3.2                                                                                                           
DEBUG: Changing to port directory: /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/lang/scala
DEBUG: OS darwin/21.5.0 (macOS 12.4) arch arm
DEBUG: Sourcing PortGroup github 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/github-1.0.tcl
DEBUG: Sourcing PortGroup java 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/java-1.0.tcl
DEBUG: Sourcing PortGroup deprecated 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/deprecated-1.0.tcl
DEBUG: Re-registering default for livecheck.url
DEBUG: Re-registering default for livecheck.regex
DEBUG: Sourcing PortGroup select 1.0 from /opt/local/var/macports/sources/rsync.macports.org/macports/release/tarballs/ports/_resources/port1.0/group/select-1.0.tcl
DEBUG: universal_variant is false, so not adding the default universal variant
DEBUG: Running callback portconfigure::add_automatic_compiler_dependencies
DEBUG: Finished running callback portconfigure::add_automatic_compiler_dependencies
DEBUG: Running callback portbuild::add_automatic_buildsystem_dependencies
DEBUG: Finished running callback portbuild::add_automatic_buildsystem_dependencies
DEBUG: Running callback portstartupitem::add_notes
DEBUG: Finished running callback portstartupitem::add_notes
DEBUG: Running callback java::java_set_env
DEBUG: java-portgroup: Trying to find JVM version: 1.8+
DEBUG: Redefining java fallback openjdk8 to openjdk8-zulu for arm compatibility
DEBUG: No value for java JAVA_HOME was automatically discovered
DEBUG: Adding dependency on JDK fallback openjdk8-zulu
DEBUG: Finished running callback java::java_set_env
DEBUG: Running callback deprecated.deprecate_port
DEBUG: Finished running callback deprecated.deprecate_port
Full Name: scala3.2 @3.2.0_1
Library Dependencies: port:scala_select, port:openjdk8-zulu
DEBUG: Checking time since last reclaim run
catap@Kirills-mini-m1 ~ % 
```

**Comment by breun** [comment]
I am not very familiar with the `java` portgroup, but support for building on ARM64 is only available for the `openjdk17` and higher ports at the moment, because it was added to vanilla OpenJDK only in OpenJDK 16. The Azul Zulu and Amazon Corretto distributions of OpenJDK also provide ARM64 builds for older versions of Java, with Azul covering more Java versions, so I think this makes sense.

**Review by breun — APPROVED** [review]

**Comment by catap** [comment]
@breun yeah, special that I'm a the man who brings `zulu` java to MacPorts for this kind of reason.

**Comment by catap** [comment]
@mascguy what do you think about this one? :)

File: _resources/port1.0/group/java-1.0.tcl [hunk 2]
```diff
@@ -59,7 +59,9 @@ namespace eval java {
             # as required, as currently these are the only ones supporting arm.
             # To be reviewed as support for arm comes for the other versions.
             # Following regex matches openjdk<version> only.
-            if { [option configure.build_arch] eq "arm64" &&
+            # Keep in mind that configure.build_arch isn't available here
+            global os.arch
+            if { ${os.arch} eq "arm" &&
```
Comment (mascguy) [inline _resources/port1.0/group/java-1.0.tcl:64]: While `[option configure.build_arch]` looks incorrect, and needs to be changed... `configure.build_arch` isn't accessible, after declaring as a global...?

Comment (catap) [reply _resources/port1.0/group/java-1.0.tcl:64]: @mascguy yes, it doesn't. It was a while until I've found `os.arch` :)

**Review by mascguy — APPROVED** [review]

**Pull request merged**